tests/matrix_free/point_evaluation_32.cc

```diff
@@ -0,0 +1,167 @@
+  {}
+
+  double
+  value(const Point<dim> &p, const unsigned int component) const override
+  {
+    return component + p[component];
+  }
+};
+
+
+
+template <int dim, int spacedim>
+void
+test(const unsigned int degree)
+{
+  Triangulation<dim, spacedim> tria;
+
+
+  if constexpr (dim + 1 == spacedim)
+    GridGenerator::hyper_sphere(tria, Point<spacedim>(), 1.);
+  else if constexpr (dim == 1)
+    GridGenerator::hyper_rectangle(tria, Point<dim>(), Point<dim>(1.2));
+  else
+    static_assert("Case not implemented");
+
+  MappingQ<dim, spacedim> mapping(degree);
+  deallog << "Mapping of degree " << degree << std::endl;
+
+  std::vector<Point<dim - 1>> unit_points;
+  for (unsigned int i = 0; i < (dim > 1 ? 13 : 1); ++i)
+    {
+      Point<dim - 1> p;
+      for (unsigned int d = 0; d < dim - 1; ++d)
+        p[d] = static_cast<double>(i) / 17. + 0.015625 * d;
+      unit_points.push_back(p);
+    }
+
+  FE_Q<dim, spacedim>       fe(degree);
+  DoFHandler<dim, spacedim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+  Vector<double> vector(dof_handler.n_dofs());
+
+  NonMatching::MappingInfo<dim, spacedim> mapping_info(mapping,
+                                                       update_values |
+                                                         update_gradients);
+
+  std::vector<std::vector<Quadrature<dim - 1>>> quad_vec_faces(
+    tria.n_active_cells());
+
+  for (const auto &cell : tria.active_cell_iterators())
+    {
+      for (const auto f : cell->face_indices())
+        {
+          quad_vec_faces[cell->active_cell_index()].push_back(
+            Quadrature<dim - 1>(unit_points));
+        }
+    }
+
+  mapping_info.reinit_faces(tria.active_cell_iterators(), quad_vec_faces);
+
+  FEFacePointEvaluation<1, dim, spacedim> evaluator(mapping_info, fe);
+
+  VectorTools::interpolate(mapping,
+                           dof_handler,
+                           MyFunction<spacedim>(),
+                           vector);
+
+  std::vector<double> solution_values_in(fe.dofs_per_cell);
+  std::vector<double> solution_values_out(fe.dofs_per_cell);
+
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      cell->get_dof_values(vector,
+                           solution_values_in.begin(),
+                           solution_values_in.end());
+
+      deallog << "Cell with center " << cell->center(true) << std::endl;
+      for (const auto f : cell->face_indices())
+        {
+          evaluator.reinit(cell->active_cell_index(), f);
+
+          evaluator.evaluate(solution_values_in,
+                             EvaluationFlags::values |
+                               EvaluationFlags::gradients);
+          deallog << "face number " << f << std::endl;
+          for (unsigned int i = 0; i < unit_points.size(); ++i)
+            deallog << "value " << evaluator.get_value(i) << " gradient "
+                    << evaluator.get_gradient(i) << std::endl;
+          deallog << std::endl;
```

Comment:
Should we also compare to `FEFaceValues` like in the other tests? How did you check if the results are correct?

Reply:
I added `FEFaceValues` now; we didn't have it in the test I copied it from, but there is no damage in adding it. We now also print the reference and see that the values are OK. I had previously just verified the part with linear function in dim=1, spacedim=1 with a rectangular domain, but of course it is good to verify the data on the sphere as well.